src/mapclient/view/dialogs/selfclosing/messagebox.py

```diff
@@ -4,6 +4,13 @@
 
 class MessageBox(QtWidgets.QMessageBox):
 
-    def __init__(self, icon, title, text, buttons, parent=None, close_after=2000):
+    close_after = None
```

Comment:
Why has this been placed here? and not in the __init__ function?

Reply:
I had made these updates under the assumption that we may eventually want to use the MessageBox class for messages other than _just_ notifying the user of a successful workflow completion.

I thought that having a _class_ variable rather than an _instance_ variable would be the simplest way to ensure that all instances of the MessageBox class would be displayed for the user-specified length of time (the value on the _Options_ dialog). Otherwise, we would have to read the timer value from the OptionsManager each time we create a MessageBox object.

Maybe I'm missing something...

Reply:
No, I don't think that you are missing something. But, what we have now is only the one message box with no clear path forward for adding additional message boxes and nothing for time limited messages boxes expected anytime soon. So, under these circumstances I think it is better to query the option on message box creation.  Doing this will in no way effect the performance of displaying the message box.

Reply:
Yep. Makes sense. I've made another commit that changes the close_after class variable to an instance variable.